### PR TITLE
Enable bancor for requests with BNT

### PR DESCRIPTION
**Pull request by xianny**
# Description

Enable Bancor as a source, but only for requests with BNT.

# Testing Instructions

# Checklist

-   [ ] Update [documentation](https://github.com/0xProject/website/blob/development/mdx/api/index.mdx) as needed. **Website Documentation PR:**
-   [ ] Prefix PR title with `[WIP]` if necessary.
-   [ ] Add tests to cover changes as needed.

### CODE REVIEW

**Comment by xianny** [comment]
deploy staging

**Comment by xianny** [comment]
deploy staging

**Comment by xianny** [comment]
deploy staging

File: src/handlers/meta_transaction_handlers.ts [hunk 1]
```diff
@@ -365,10 +365,14 @@ const parseGetTransactionRequestParams = (req: express.Request): GetTransactionR
...
         ]);
     }
-    const excludedSources =
+    const _excludedSources =
         req.query.excludedSources === undefined
             ? undefined
             : parseUtils.parseStringArrForERC20BridgeSources((req.query.excludedSources as string).split(','));
+    // Exclude Bancor as a source unless swap involves BNT token
+    const isBNT = sellToken.toUpperCase() === 'BNT' || buyToken.toUpperCase() === 'BNT';
+    const excludedSources = isBNT ? _excludedSources : _excludedSources.concat(ERC20BridgeSource.Bancor);
```
Comment (kimpers) [inline src/handlers/meta_transaction_handlers.ts:374]: Would this still work if `sellToken` or `buyToken` is provided as an address in the request?

Comment (xianny) [reply src/handlers/meta_transaction_handlers.ts:374]: good catch, changed it to filter with both

**Review by kimpers — APPROVED** [review]
LGTM 👍

**Pull request merged**